### PR TITLE
[VC02-4326] Sanity checks for variants + refactoring

### PR DESCRIPTION
**Pull request by trojkat**
- sanity checks for variants in timesloted and non-timesloted products
- merging similar tests for 30 days availability into a single test case
- fix for handling `order_reference` in the mock server

### CODE REVIEW

File: supplier_api_tester/supplier_api_tester/tests/availability_timeslots.py [hunk 1]
```diff
@@ -1,39 +1,96 @@
...
+    unique_dates = set(dates)
+    if len(unique_dates) == len(dates):
+        unique_starts_ends = {f'{t.start}-{t.end}' for t in timeslots}
+        if len(unique_starts_ends) == 1:
+            raise FailedTest(
+            message='If a product contains only a single timeslot at the same time every day, then please implement it as a non-timesloted product',
+            response=raw_response,
+            )
+
+    # Checking if timslots are uniqe for each day
```
Comment (kajetanjuszczak) [inline supplier_api_tester/supplier_api_tester/tests/availability_timeslots.py:83]: timeslots

File: supplier_api_tester/supplier_api_tester/utils/reservation.py [hunk 2]
```diff
@@ -4,25 +4,20 @@
...
 from supplier_api_tester.exceptions import FailedTest
 from supplier_api_tester.models import DailyVariants, Timeslot
-from supplier_api_tester.utils.adapters import get_variants, get_timeslots
+from supplier_api_tester.utils.availability import (
+    get_availability_timeslots,
+    get_availability_variants,
+)
 
 
 def get_reservation_slot(api_url, api_key, product_id, timeslots: bool, version=1):
```
Comment (kajetanjuszczak) [inline supplier_api_tester/supplier_api_tester/utils/reservation.py:13]: return type - dict I guess

**Review by kajetanjuszczak — APPROVED** [review]

File: supplier_api_tester/supplier_api_tester/tests/availability_timeslots.py [hunk 3]
```diff
@@ -1,39 +1,96 @@
...
+        api_url,
+        api_key,
+        product_id,
+        version,
+        start_date,
+        end_date,
+    )
+
+    # non-zero availability check
+    max_tickets_sum = sum(timeslot.max_tickets for timeslot in timeslots)
```
Comment (Loreq12) [inline supplier_api_tester/supplier_api_tester/tests/availability_timeslots.py:27]: I think this along with variant uniquness is duplicated in 30 days check. Is it worth to extract this?

Comment (trojkat) [reply supplier_api_tester/supplier_api_tester/tests/availability_timeslots.py:27]: I left it only to show that we're iterating over different objects.

**Review by Loreq12 — APPROVED** [review]

**Pull request merged**